app/src/main/java/org/mozilla/fenix/compose/PrimaryAction.kt

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Button
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Button for homepage items.
+ *
+ * @param text button text.
+ * @param icon button icon (if any).
+ * @param onClick Invoked when button is clicked.
+ */
+@Composable
+fun PrimaryAction(
+    text: String,
+    icon: Painter?,
+    onClick: () -> Unit,
+) {
+    Button(
+        onClick = onClick,
+        modifier = Modifier
+            .clip(RoundedCornerShape(size = 4.dp))
+            .fillMaxWidth()
+            .height(36.dp),
+        elevation = ButtonDefaults.elevation(defaultElevation = 0.dp, pressedElevation = 0.dp),
+        colors = ButtonDefaults.outlinedButtonColors(backgroundColor = FirefoxTheme.colors.actionPrimary),
+    ) {
+        if (icon != null) {
+            Icon(
+                painter = icon,
+                contentDescription = null,
+                tint = FirefoxTheme.colors.iconOnColor,
+            )
+
+            Spacer(modifier = Modifier.width(8.dp))
+        }
+
+        Text(
+            text = text,
+            modifier = Modifier.align(Alignment.CenterVertically),
+            color = FirefoxTheme.colors.textOnColorPrimary,
+            fontSize = 14.sp,
+            fontWeight = FontWeight.SemiBold,
+            letterSpacing = 0.sp,
```

Comment:
I imagine we don't need to specify the `letterSpacing` if it's 0.sp

Reply:
There is actually spacing present in the default text, hence to remove that we added this. This was provided in the NeutralButton parent of PositiveButton style that we inherit in the xml file.

```
<style name="NeutralButton" parent="Widget.MaterialComponents.Button.TextButton">
        <item name="iconTint">@color/fx_mobile_icon_color_action_secondary</item>
        <item name="iconPadding">8dp</item>
        <item name="iconGravity">textStart</item>
        <item name="android:textAlignment">center</item>
        <item name="android:layout_width">match_parent</item>
        <item name="android:radius">4dp</item>
        <item name="android:layout_height">48dp</item>
        <item name="android:textStyle">bold</item>
        <item name="android:textAllCaps">false</item>
        <item name="backgroundTint">@color/fx_mobile_action_color_secondary</item>
        <item name="android:textColor">@color/fx_mobile_text_color_action_secondary</item>
        <item name="rippleColor">?attr/textSecondary</item>
        <item name="android:letterSpacing">0</item>
        <item name="fontFamily">@font/metropolis_semibold</item>
    </style>
```